session/src/main/scala/akka/stream/alpakka/cassandra/AkkaDiscoverySessionProvider.scala

```diff
@@ -47,65 +42,50 @@ import scala.concurrent.{ ExecutionContext, Future }
+private[cassandra] object AkkaDiscoverySessionProvider {
 
-  /**
-   * Use Akka Discovery to read the addresses for `serviceName` within `lookupTimeout`.
-   */
-  private def readNodes(serviceName: String, lookupTimeout: FiniteDuration)(
-      implicit system: ActorSystem): Future[immutable.Seq[String]] = {
-    Discovery(system).discovery
-      .lookup(serviceName, lookupTimeout)
-      .map { resolved =>
-        resolved.addresses.map(target => target.host + ":" + target.port)
-      }(ExecutionContexts.sameThreadExecutionContext)
+  def connect(system: ActorSystem, config: Config)(implicit ec: ExecutionContext): Future[CqlSession] = {
+    readNodes(config)(system).flatMap { contactPoints =>
+      val driverConfigWithContactPoints = ConfigFactory.parseString(s"""
+        basic.contact-points = [${contactPoints.mkString("\"", "\", \"", "\"")}]
+        """).withFallback(CqlSessionProvider.driverConfig(system, config))
+      val driverConfigLoader = DriverConfigLoaderFromConfig.fromConfig(driverConfigWithContactPoints)
+      CqlSession.builder().withConfigLoader(driverConfigLoader).buildAsync().toScala
+    }
   }
 
   /**
    * Expect a `service` section in Config and use Akka Discovery to read the addresses for `name` within `lookup-timeout`.
    */
   private def readNodes(config: Config)(implicit system: ActorSystem): Future[immutable.Seq[String]] = {
-    val serviceConfig = config.getConfig("service")
+    val serviceConfig = config.getConfig("service-discovery")
     val serviceName = serviceConfig.getString("name")
     val lookupTimeout = serviceConfig.getDuration("lookup-timeout").asScala
     readNodes(serviceName, lookupTimeout)
   }
 
-  override def connect()(implicit ec: ExecutionContext): Future[CqlSession] = {
-    val driverConfig = CqlSessionProvider.driverConfig(system, config)
-    val sessionName =
-      if (driverConfig.hasPath("basic.session-name")) driverConfig.getString("basic.session-name")
-      else ""
-    require(sessionName != null && sessionName.nonEmpty, s"driver config needs to set a non-empty `basic.session-name`")
-
-    readNodes(config)(system).flatMap { contactPoints =>
-      val driverConfigWithContactPoints = ConfigFactory.parseString(s"""
-        basic.contact-points = [${contactPoints.mkString("\"", "\", \"", "\"")}]
-        """).withFallback(driverConfig)
-
-      val driverConfigLoader = DriverConfigLoaderFromConfig.fromConfig(driverConfigWithContactPoints)
-      CqlSession.builder().withConfigLoader(driverConfigLoader).buildAsync().toScala
-    }
+  /**
+   * Use Akka Discovery to read the addresses for `serviceName` within `lookupTimeout`.
+   */
+  private def readNodes(serviceName: String, lookupTimeout: FiniteDuration)(
+      implicit system: ActorSystem): Future[immutable.Seq[String]] = {
+    Discovery(system).discovery
+      .lookup(serviceName, lookupTimeout)
+      .map { resolved =>
+        resolved.addresses.map(target => target.host + ":" + target.port)
```

Comment:
does it always have a port or do we need to fallback to default port (or not include the port part)

Reply:
Thank you for spotting this.
Cassandra reports `Ignoring invalid contact point 127.0.0.1 (expecting host:port)` when the port is missing.
New handling and test case added.